railties/test/application/rake_test.rb

```diff
@@ -122,10 +122,43 @@ def test_rake_routes_with_controller_environment
       RUBY
 
       ENV['CONTROLLER'] = 'cart'
+
+      assert_deprecated { Dir.chdir(app_path){ `bin/rake routes` } }
```

Comment:
Prefer initializing `output` outside the block, so we don't call the rake task twice:

```ruby
output = nil
assert_deprecated { output = Dir.chdir(app_path) { `bin/rake routes` } }
```